### ISSUE TITLE
Invalid twig syntax + grammar

### ISSUE THREAD

**Issue by nicoSWD** [dddinphp/ddd-in-php-book-issues#49]
In section **2.2.2.2 The View**, it should read:

``` php
{% endfor %}
```

Instead of:

``` php
{% endforeach %}
```

**2.3.1 The Dependency Inversion Principle (DIP)**

This:

> But then, what is Hexagonal Architecture?, and how does it fit within of all this?

Should read:

> But then, what is Hexagonal Architecture, and how does it fit within ~~of~~ all **_of_** this?

Or:

> But then, what is Hexagonal Architecture? And how does it fit within ~~of~~ all **_of_** this?

**2.3.2 Applying Hexagonal Architecture**

``` php
$post = Post::writeNewFrom($title, $content);
```

Maybe it's just me, but the naming feels a little weird. Why does it appear to "write" a new "form", but is returning an instance of  `Post`? Wouldn't `Post::create`, or as of PHP 7, `Post::new` be more appropriate?

**2.4 Command Query Responsibility Segregation**

> This strict separation triggers another problem, Eventual Consistency.

I would rephrase this to:

> This strict separation causes another problem: Eventual Consistency.

You used the word "triggers" in the sentence prior to this one, and it feels a little out of place here.

And that's it for today. Will continue reading in the upcoming days. I understand you're having the book professionally reviewed, so should I keep focusing on things like grammar, or should I concentrate mainly on the technical parts?

**Comment by theUniC** [comment]
Thanks Nico! :)

I've fixed all except the `Post::writeNewFrom`.

To my understand, the name `writeNewFrom` is much more aligned with the **Ubiquitous Language** than `create` or simply `new`, since when you want to publish a blog post you have to first _write it_. The Blog Post example it's maybe a simple example, but I think that named constructors should strictly follow the Ubiquitous Language.

Christian.

**Comment by nicoSWD** [comment]
Thanks for the explanation, Christian! Fair enough, I'll accept that as an answer. :)
